### PR TITLE
 Setup all the stores required to store headers and aggregateCommits - Closes #7220 

### LINKED ISSUE #7220: Setup all the stores required to store headers and aggregateCommits
### Description

- Implement `getDBInstance`
- Implement `getChainConnectorInfo`
- Implement `setChainConnectorInfo(ChainConnectorInfo)`
- Add `DB_KEY_CHAIN_CONNECTOR_INFO` key
- Use `EMPTY_BUFFER` as key
#### Data structure
Data structure to be saved in the `ChainConnectorDB`

```ts
interface ChainConnectorInfo {
   blockHeader: BlockHeader[],
   aggregateCommits: AggregateCommit[],
   validatorsHashPreimage: ValidatorsData[],
}
```

**Schemas**
- AggregateCommit schema: [AggregateCommit Schema](https://github.com/LiskHQ/lisk-sdk/blob/e98d0ecd7aa67cbbd3ed35199c9ea99f7d31d594/framework/src/engine/consensus/certificate_generation/schema.ts#L96)
- BlockHeader Schema: Import from SDK `chain.blockHeaderSchema`
- ValidatorsData Schema: [ValidatorsData Schema](https://github.com/LiskHQ/lisk-sdk/blob/e4b4fbc064ea450ad21d72a46062507d704aac0d/framework/src/engine/bft/schemas.ts#L25)

### Acceptance Criteria

- All the unit tests testing each of these functions should exist

### Additional Info

- Depends on #7218

### PR DESCRIPTION
**Pull request by Incede**
### What was the problem?

This PR resolves #7220

### How was it solved?

Add functions as specified

### How was it tested?

Added unit test

### CODE REVIEW

File: framework-plugins/lisk-framework-chain-connector-plugin/src/schemas.ts [hunk 1]
```diff
@@ -39,3 +41,26 @@ export const configSchema = {
...
+			},
+		},
+		aggregateCommits: {
+			type: 'object',
+			fieldNumber: 2,
+		},
+		validatorsHashPreimage: {
+			type: 'object',
+			fieldNumber: 3,
+		},
```
Comment (ishantiw) [inline framework-plugins/lisk-framework-chain-connector-plugin/src/schemas.ts:63]: I think we need to expand these

File: framework-plugins/lisk-framework-chain-connector-plugin/src/db.ts [hunk 2]
```diff
@@ -11,3 +11,41 @@
...
+	dataPath: string,
+	dbName = 'lisk-framework-chain-connector-plugin.db',
+): Promise<KVStore> => {
+	const dirPath = join(dataPath.replace('~', os.homedir()), 'plugins/data', dbName);
+	await ensureDir(dirPath);
+
+	return new Database(dirPath);
+};
+
+export const getChainConnectorInfo = async (db: KVStore): Promise<ChainConnectorInfo> => {
```
Comment (ishantiw) [inline framework-plugins/lisk-framework-chain-connector-plugin/src/db.ts:39]: I missed to mention to implement `setChainConnectorInfo(ChainConnectorInfo)`. I have updated the issue

**Review by ishantiw — CHANGES_REQUESTED** [review]
Overall LGTM, just a couple of requested changes

File: framework-plugins/lisk-framework-chain-connector-plugin/src/schemas.ts (hunk 1 above)
Comment (Incede) [reply framework-plugins/lisk-framework-chain-connector-plugin/src/schemas.ts:63]:
```	aggregateCommits: Record<number, AggregateCommit>;
	validatorsHashPreimage: Record<string, ValidatorsData>;
```
Can you maybe give an example how to expand them ?

Comment (ishantiw) [reply framework-plugins/lisk-framework-chain-connector-plugin/src/schemas.ts:63]: as we cannot use (key, value) pair in codec unless we create a substore. we can use array of [aggregateCommit Schema](https://github.com/LiskHQ/lisk-sdk/blob/e98d0ecd7aa67cbbd3ed35199c9ea99f7d31d594/framework/src/engine/consensus/certificate_generation/schema.ts#L96)

validatorsHashPreimage we can use array of 

```ts
validatorsData {
        prevoteThreshold: bigint;
	precommitThreshold: bigint;
	certificateThreshold: bigint;
	validators: BFTValidator[];
	validatorsHash: Buffer;
} []
```
which we are getting from `getBFTParameters` function in BFT

File: framework-plugins/lisk-framework-chain-connector-plugin/src/db.ts (hunk 2 above)
Comment (Incede) [reply framework-plugins/lisk-framework-chain-connector-plugin/src/db.ts:39]: The set function always needs a key value pair as input
```
Implement setChainConnectorInfo(ChainConnectorInfo)
```
But the updated issue only has a single input parameter.

File: framework-plugins/lisk-framework-chain-connector-plugin/src/schemas.ts (hunk 1 above)
Comment (Incede) [reply framework-plugins/lisk-framework-chain-connector-plugin/src/schemas.ts:63]: Thank you, can you maybe also formulate a example in terms of exact schema for atleast one of them like
```
	aggregateCommits: {
			type: 'object',

			fieldNumber: 2,
		},
		validatorsHashPreimage: {
			type: 'object',
			fieldNumber: 3,
		},
```

File: framework-plugins/lisk-framework-chain-connector-plugin/src/schemas.ts [hunk 3]
```diff
@@ -42,6 +42,28 @@ export const configSchema = {
 	},
 };
 
+export const validatorsDataSchema = {
+	$id: '/modules/bft/validatorsHashInput',
+	type: 'object',
+	required: ['activeValidators', 'certificateThreshold'],
+	properties: {
```
Comment (ishantiw) [inline framework-plugins/lisk-framework-chain-connector-plugin/src/schemas.ts:49]: `validatorsHash` is missing, we want to store where we can leave `precommitThreshold` and `prevoteThreshold` https://github.com/LiskHQ/lisk-sdk/blob/development/framework/src/engine/bft/schemas.ts#L25

**Review by ishantiw — CHANGES_REQUESTED** [review]
Looks good, just one ,missing property on validatorsData

**Review by ishantiw — APPROVED** [review]
🎉

**Review by has5aan — APPROVED** [review]

**Pull request merged**